python/graphstorm/eval/evaluator.py

```diff
@@ -905,27 +669,46 @@ def compute_score(self, pred, labels):
             Evaluation metric values: dict
         """
         scores = {}
-        if pred is None or labels is None:
-            for metric in self.metric:
+        for metric in self.metric_list:
+            if pred is not None and labels is not None:
+                pred = th.squeeze(pred)
+                labels = th.squeeze(labels)
+                pred = pred.to(th.float32)
+                labels = labels.to(th.float32)
+
+                if train:
+                    # training expects always a single number to be
+                    # returned and has a different (potentially) evluation function
+                    scores[metric] = self.metrics_obj.metric_function[metric](pred, labels)
```

Comment:
I do not quite understand the design here? Who is going to set metric_function and metric_eval_function?

Reply:
Yes, in this way, we allow users to set different eval function for train=True or False.